lib/galaxy/security/validate_user_input.py

```diff
@@ -45,7 +45,7 @@ def validate_publicname( trans, publicname, user=None ):
         if len( publicname ) < 3:
             return "Public name must be at least 3 characters in length"
     else:
-        if len( publicname ) < 4:
+        if len( publicname ) < 3:
             return "Public name must be at least 4 characters in length"
```

Comment:
If 3 is ok then doesn't need separate logic for tool shed and I'll tidy that. Was in a rush to push a PR to get some initial comments on direction as have had to switch to a different project for a bit. Will do some tidying at home. Thanks.